### PR TITLE
Initial hardware documentation for OQC

### PR DESCRIPTION
**Pull request by owen-oqc**
OQC target general use documentation

### CODE REVIEW

File: docs/sphinx/using/hardware.rst [hunk 1]
```diff
@@ -308,3 +308,87 @@ the ``shots_count`` is set to 1000.
...
+    nvq++ --emulate --target oqc src.cpp -o executable
+
+
+.. note::
+
+    There is currently no requirement to specify the machine in use. This is configured via the url which points to the relevant system.
+
+.. note::
+
+    The OQC quantum assembly toolchain (qat) which we use to compile and execute instructions can be found on github as `oqc-community/qat <https://github.com/oqc-community/qat>`__
```
Comment (bmhowe23) [inline docs/sphinx/using/hardware.rst:363]: Suggest replacing the "we"...
```suggestion
    The OQC quantum assembly toolchain (qat) which is used to compile and execute instructions can be found on github as `oqc-community/qat <https://github.com/oqc-community/qat>`__
```

File: docs/sphinx/using/hardware.rst [hunk 2]
```diff
@@ -308,3 +308,87 @@ the ``shots_count`` is set to 1000.
...
+1. ``OQC_URL``
+2. ``OQC_EMAIL``
+3. ``OQC_PASSWORD`` - is mandatory
+
+Submission from C++
+`````````````````````````
+
+To target quantum kernel code for execution on the OQC platform, provide the flag ``--target oqc`` to the ``nvq++`` compiler.
+
+users may provide their :code:`email` and :code:`url` as extra arguments
```
Comment (bmhowe23) [inline docs/sphinx/using/hardware.rst:336]:
```suggestion
Users may provide their :code:`email` and :code:`url` as extra arguments
```

File: docs/sphinx/using/hardware.rst [hunk 3]
```diff
@@ -308,3 +308,87 @@ the ``shots_count`` is set to 1000.
...
+
+.. code:: bash
+
+    nvq++ --target oqc --oqc-email $email --oqc-url $url src.cpp -o executable
```
Comment (bmhowe23) [inline docs/sphinx/using/hardware.rst:340]:
```suggestion
    nvq++ --target oqc --oqc-email <email> --oqc-url <url> src.cpp -o executable
```

File: docs/sphinx/using/hardware.rst [hunk 4]
```diff
@@ -308,3 +308,87 @@ the ``shots_count`` is set to 1000.
...
+
+where both environment variables and extra arguments are supplied, precedent is given to the extra arguments.
+To run the output, provide the runtime loaded variables and invoke the pre-built executable
+
+.. code:: bash
+
+   OQC_PASSWORD=$password ./executable
```
Comment (bmhowe23) [inline docs/sphinx/using/hardware.rst:347]:
```suggestion
   OQC_PASSWORD=<password> ./executable
```

File: docs/sphinx/using/hardware.rst [hunk 5]
```diff
@@ -308,3 +308,87 @@ the ``shots_count`` is set to 1000.
...
+
+.. code:: python
+
+   import os
+   import cudaq
+   # ...
+   os.environ['OQC_PASSWORD'] = password
+   cudaq.set_target(target="oqc", url="")
+
+you can then execute a kernel against the platform using the oqc Lucy device
```
Comment (bmhowe23) [inline docs/sphinx/using/hardware.rst:379]:
```suggestion
You can then execute a kernel against the platform using the OQC Lucy device
```

File: docs/sphinx/using/hardware.rst [hunk 6]
```diff
@@ -308,3 +308,87 @@ the ``shots_count`` is set to 1000.
...
+
+To target quantum kernel code for execution on the OQC platform, provide the flag ``--target oqc`` to the ``nvq++`` compiler.
+
+users may provide their :code:`email` and :code:`url` as extra arguments
+
+.. code:: bash
+
+    nvq++ --target oqc --oqc-email $email --oqc-url $url src.cpp -o executable
+
+where both environment variables and extra arguments are supplied, precedent is given to the extra arguments.
```
Comment (bmhowe23) [inline docs/sphinx/using/hardware.rst:342]:
```suggestion
When both environment variables and extra arguments are supplied, precedence is given to the extra arguments.
```

**Review by bmhowe23 — COMMENTED** [review]
Thanks for pulling this together. I just have a few very minor suggestions below.

**Comment by owen-oqc** [comment]
> Thanks for pulling this together. I just have a few very minor suggestions below.

Thanks @bmhowe23 suggestions adopted.

**Comment by bmhowe23** [comment]
I plan on bringing this one in after #838 is merged.

File: docs/sphinx/using/hardware.rst (hunk 1 above)
Comment (bmhowe23) [reply docs/sphinx/using/hardware.rst:363]: I believe this previously applied suggestion may have been lost as part of the recent force pushes.

File: docs/sphinx/using/hardware.rst (hunk 3 above)
Comment (bmhowe23) [reply docs/sphinx/using/hardware.rst:340]: I believe this previously applied suggestion may have been lost as part of the recent force pushes.

File: docs/sphinx/using/hardware.rst (hunk 4 above)
Comment (bmhowe23) [reply docs/sphinx/using/hardware.rst:347]: I believe this previously applied suggestion may have been lost as part of the recent force pushes.

File: docs/sphinx/using/hardware.rst (hunk 5 above)
Comment (bmhowe23) [reply docs/sphinx/using/hardware.rst]: I believe this previously applied suggestion may have been lost as part of the recent force pushes.

File: docs/sphinx/using/hardware.rst (hunk 6 above)
Comment (bmhowe23) [reply docs/sphinx/using/hardware.rst]: I believe this previously applied suggestion may have been lost as part of the recent force pushes.

**Pull request merged**